### PR TITLE
Accounts app created. 

### PR DESCRIPTION
**Pull request by codeTony22**
Accounts app takes care of registering the user, log in, log out, profile view, editing profile, and password reset. Functionality is working, I will polish it more during coming springs.

### CODE REVIEW

File: webapp/accounts/static/accounts/style.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+h1 {
```
Comment (loksonarius) [inline webapp/accounts/static/accounts/style.css:1]: Any reason this is still here?
Global CSS changes should go in the static files' dir of the `geek_text` app.

File: webapp/accounts/forms.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+from django.contrib.auth.models import User
+from django.contrib.auth.forms import UserCreationForm, UserChangeForm
+from django import forms
+
+
+class RegistrationForm(UserCreationForm):
+    email = forms.EmailField(required=True)
```
Comment (loksonarius) [inline webapp/accounts/forms.py:7]: This had to be manually added despite the `User` model having an `email` field? Thought that'd be added automatically 🤔

File: webapp/geek_text/settings.py [hunk 3]
```diff
@@ -122,3 +123,4 @@
 # https://docs.djangoproject.com/en/1.11/howto/static-files/
 
 STATIC_URL = '/static/'
+LOGIN_REDIRECT_ULR='/account/'
```
Comment (loksonarius) [inline webapp/geek_text/settings.py:136]: This constant seems to be misspelled.
Please fix this; the end part of it is `ULR` instead of `URL`.

File: webapp/accounts/urls.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+from django.contrib.auth.views import (
+    login, logout,
+    password_reset, password_change_done,
+    password_reset_confirm
+)
+
+
+urlpatterns = [
+    url(r'^$', views.home),
+    url(r'^login/$', login , {'template_name': 'accounts/login.html'}, name='login'),
```
Comment (loksonarius) [inline webapp/accounts/urls.py:12]: These regex's shouldn't need a trailing `/`.
URL's in terms of RESTful resources shouldn't end with trailing `/`s.
Please remove them from resources that don't need them.

File: webapp/accounts/templates/base.html [hunk 5]
```diff
@@ -0,0 +1,44 @@
+<!DOCTYPE html>
```
Comment (loksonarius) [inline webapp/accounts/templates/base.html:1]: We already have a base.html template in `master` under the global app.
We do not need to duplicate it here; please delete this and adjust the code accordingly.

File: webapp/accounts/urls.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+from django.conf.urls import url
+from .import views
+from django.contrib.auth.views import (
```
Comment (loksonarius) [inline webapp/accounts/urls.py:3]: Why are we importing views from the `auth` app?
Why is this needed here? It seems like we could have just had a login, logout and profile page to suffice.

**Review by loksonarius — CHANGES_REQUESTED** [review]
It is currently looking like way too much was implemented in one go.
We may want to split this up into smaller groups of work to ensure good code.

**Comment by loksonarius** [comment]
It would be good to run this code through Pylint to clean it up a bit; the style is quite different from what's currently in `master`.

File: webapp/accounts/models.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
+from django.contrib.auth.models import User
+from django.db import models
+from django.db.models.signals import post_save
+
+
+# Create your models here.
+class UserProfile(models.Model):
```
Comment (loksonarius) [inline webapp/accounts/models.py:7]: It is a good idea to have a separate model for additional User details, but it doesn't seem like it is actually being used.
I don't see any forms for this model either.

File: webapp/accounts/templates/accounts/change_password.html [hunk 8]
```diff
@@ -0,0 +1,16 @@
+{% extends 'base.html' %}
+
+{%block head%}
+<title>Profile</title>
```
Comment (loksonarius) [inline webapp/accounts/templates/accounts/change_password.html:4]: This file looks like a duplicate of the `edit_profile` template.
Why do we need this file?

File: webapp/geek_text/urls.py [hunk 9]
```diff
@@ -15,10 +15,15 @@
 """
 from django.conf.urls import url, include
 from django.contrib import admin
+from .views import login_redirect
 
 urlpatterns = [
+    url(r'^$', login_redirect , name='login_redirect'),
```
Comment (loksonarius) [inline webapp/geek_text/urls.py:21]: Why are we forcing users to log in right off the bat when they hit the website?
This isn't favorable UX for a shopping site.

File: webapp/geek_text/urls.py [hunk 10]
```diff
@@ -15,10 +15,15 @@
...
     url(r'^admin/', admin.site.urls),
+    url(r'^accounts/', include('accounts.urls')),
     url(r'^browse/', include('browse.urls', namespace='browse')),
     url(r'^books/', include('book_details.urls', namespace='books')),
     url(r'^reviews/', include('reviews.urls', namespace='reviews')),
 ]
+
```
Comment (loksonarius) [inline webapp/geek_text/urls.py:28]: Please delete the excessive newlines here.

File: webapp/geek_text/views.py [hunk 11]
```diff
@@ -0,0 +1,4 @@
+from django.shortcuts import redirect
```
Comment (loksonarius) [inline webapp/geek_text/views.py:1]: We shouldn't have logic like this from the `accounts` app leak into the global app. Redirects to log in forms are really only needed when trying to edit profiles or finalize a purchase.
Please delete this.

File: webapp/geek_text/urls.py [hunk 12]
```diff
@@ -15,10 +15,15 @@
 """
 from django.conf.urls import url, include
 from django.contrib import admin
+from .views import login_redirect
```
Comment (loksonarius) [inline webapp/geek_text/urls.py:18]: Please see my comment about this view.

File: webapp/geek_text/settings.py (hunk 3 above)
Comment (loksonarius) [reply webapp/geek_text/settings.py:136]: I'm also not sure why we need this redirect defined here.

File: webapp/accounts/views.py [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+        if form.is_valid():
+            form.save()
+            return redirect('/account/profile')
+    else:
+        form = EditProfileForm(instance=request.user)
+        args = {'form': form}
+        return render(request, 'accounts/edit_profile.html', args)
+
+@login_required
+def change_password(request):
```
Comment (loksonarius) [inline webapp/accounts/views.py:62]: I'm not sure how a password change is different from a profile change.
These look like they should be consolidated.

File: webapp/accounts/views.py [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+    EditProfileForm
+)
+from django.shortcuts import render, redirect
+from django.contrib.auth.forms import PasswordChangeForm
+from django.contrib.auth import update_session_auth_hash
+from django.contrib.auth.decorators import login_required
+
+
+# Create your views here.
+'''Home view
```
Comment (loksonarius) [inline webapp/accounts/views.py:12]: We shouldn't need a "Home" view for an individual app like this.
Please remove.

File: webapp/accounts/urls.py [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+    login, logout,
+    password_reset, password_change_done,
+    password_reset_confirm
+)
+
+
+urlpatterns = [
+    url(r'^$', views.home),
```
Comment (loksonarius) [inline webapp/accounts/urls.py:11]: Please see comment regarding a "Home" view.

File: webapp/accounts/urls.py [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+    url(r'^login/$', login , {'template_name': 'accounts/login.html'}, name='login'),
+    url(r'^logout/$', logout , {'template_name': 'accounts/logout.html'}, name='logout'),
+    url(r'^register/$' , views.register, name='register'),
+    url(r'^profile/$', views.view_profile, name='profile' ),
+    url(r'^profile/edit/$', views.edit_profile,
+        name='edit_profile'),
+    url(r'^change_password/$', views.change_password,
```
Comment (loksonarius) [inline webapp/accounts/urls.py:18]: I don't believe we need this at all.

File: webapp/accounts/templates/accounts/login.html [hunk 17]
```diff
@@ -0,0 +1,20 @@
+{% extends 'base.html' %}
```
Comment (loksonarius) [inline webapp/accounts/templates/accounts/login.html:1]: The templates for `login` and `logout` look like standalone pages, not forms.
In the URLs for this app, they are passed in as forms.
I'm not sure this is how the views being used are meant to work.

File: webapp/accounts/static/accounts/style.css (hunk 1 above)
Comment (codeTony22) [reply webapp/accounts/static/accounts/style.css:1]: Yes, I will remove that. There is no reason for that to be there

File: webapp/accounts/models.py (hunk 7 above)
Comment (loksonarius) [reply webapp/accounts/models.py:7]: If keeping this model, please update it's fields to match the style used in our other apps.
Also, please make some use of this model; otherwise, please remove it entirely.

File: webapp/accounts/forms.py (hunk 2 above)
Comment (codeTony22) [reply webapp/accounts/forms.py:7]: I thought it was to make the email field a requirement when filling out the form

File: webapp/geek_text/settings.py (hunk 3 above)
Comment (codeTony22) [reply webapp/geek_text/settings.py:136]: Will do

**Review by loksonarius — CHANGES_REQUESTED** [review]
Please remove the .idea directories 😢
